### PR TITLE
Encryption: do not expose key

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #8351 

Problem Summary:

### What is changed and how it works?

`FileEncryptionInfo` private inherite `FileEncryptionInfoRaw`, not expose key. 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/KVStore/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,7 @@ add_headers_and_sources(kvstore ./Read)
 add_subdirectory (./MultiRaft/Disagg)
 
 add_library(kvstore ${kvstore_headers} ${kvstore_sources})
+target_include_directories(kvstore PRIVATE ${TiFlash_SOURCE_DIR}/contrib/GmSSL/include)
```
Comment (CalvinNeo) [inline dbms/src/Storages/KVStore/CMakeLists.txt:27]: Do we use `USE_GM_SSL` here?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/KVStore/CMakeLists.txt:27]: work like this: FileEncryption.h -> AESCTRCipherStream.h -> AESCTRCipher.h -> gmssl/sm4.h

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/Storages/KVStore/FFI/FileEncryption.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+BlockAccessCipherStreamPtr FileEncryptionInfo::createCipherStream(
+    const EncryptionPath & encryption_path,
+    bool is_new_created_info) const
+{
+    // If the encryption info is newly created, we should check the res to make sure the encryption is enabled.
+    if (is_new_created_info && res == FileEncryptionRes::Disabled)
+        return nullptr;
+
+    RUNTIME_CHECK_MSG(res != FileEncryptionRes::Error, "Failed to get encryption info.");
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/FFI/FileEncryption.cpp:33]: maybe we need to print the `FileEncryptionInfoRaw::error_msg`

File: dbms/src/Storages/KVStore/FFI/FileEncryption.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if (method == EncryptionMethod::Plaintext || method == EncryptionMethod::Unknown)
+        return nullptr;
+
+    const String & encryption_key = *key;
+    RUNTIME_CHECK_MSG(encryption_key.size() == DB::Encryption::keySize(method), "Encryption key size mismatch.");
+    RUNTIME_CHECK_MSG(iv->size() == DB::Encryption::blockSize(method), "Encryption iv size mismatch.");
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/FFI/FileEncryption.cpp:39]: Add `method` and the actual value that is not as expected to the exception msg

**Review by JaySon-Huang — APPROVED** [review]
LGTM with some error message comments

**Pull request merged**